### PR TITLE
retitle: add config option to retitle merged/closed PRs

### PR DESCRIPTION
**Pull request by nikhita**
Fixes https://github.com/kubernetes/test-infra/issues/13716

/assign @stevekuznetsov

### CODE REVIEW

**Comment by alvaroaleman** [comment]
I am opposed to this idea. When you squash merge, the PR title will end up being the title of the final commit. Also AFAIK even with merge merging, the merge commit gets the PRs title as title.

Being able to change the title of a PR after it got merged will offer the option to create a whole lot of confusion, we should IMO not do that.

**Comment by petr-muller** [comment]
@alvaroaleman I think relying on commit messages matching PR titles is relying on an accidental behavior. PR authors can change the title of a closed PR anyway, so you cannot really 100% prevent the confusion, and this helps larger orgs with title-based workflows (like ours with a BZ plugin).

**Comment by alvaroaleman** [comment]
May all be, but the fact that there is already a way to do bad things is not a reason to do another way. If you have a title-based workflow that currently forces you to change titles, you should fix the workflow instead of extending this plugin.

**Comment by alvaroaleman** [comment]
I mostly think it will create a lot of confusion if merged PRs suddenly have different titles than the corresponding commit. There is no automation that relies on this to work.

**Comment by nikhita** [comment]
Perhaps as a middle ground, we can add a config option for this?

**Comment by spiffxp** [comment]
I'm fine with a config option and agree it would be useful for prow.k8s.io

File: prow/plugins/config.go [hunk 1]
```diff
@@ -184,6 +185,12 @@ type RequireSIG struct {
 	GroupListURL string `json:"group_list_url,omitempty"`
 }
 
+// Retitle specifies configuration for the retitle plugin.
+type Retitle struct {
+	// AllowClosedIssues allows retitling closed/merged issues and PRs.
+	AllowClosedIssues bool `json:"allow_closed_issues,omitempty"`
```
Comment (nikhita) [inline prow/plugins/config.go]: Added a config option but words are failing me right now. I think we can come up with a better name...

Comment (alvaroaleman) [reply prow/plugins/config.go]: I think the name is fine

**Comment by nikhita** [comment]
@cblecker @stevekuznetsov can you approve if this lgty? :)

**Comment by cblecker** [comment]
/lgtm
/approve
/hold

Holding in case @stevekuznetsov has any comments

**Comment by stevekuznetsov** [comment]
... How did I miss this. Sorry! 

/hold cancel

**Pull request merged**